### ISSUE TITLE
Using ZSH on MSYS2 makes ./emsdk_env.sh run carridge return as a command

### ISSUE THREAD

**Issue by FrostKiwi** [emscripten-core/emscripten#19344]
This is a low-priority seemingly cosmetic bug, with no effects on functionality.

I have been programming with emscripten for a while now on Windwos, in the MinGW64 environment of MSYS2. I am also a `zsh` user. I have been seeing the same error for over two years now, when running `source ./emsdk_env.sh`: `(eval):1: command not found: ^M`

Strangely this only happens with `zsh`. Even without `.zshrc` present it still happens and does not happen in `bash`. Here is a video of the behavior, where I show it happening in `zsh` but not in `bash`.

https://github.com/emscripten-core/emscripten/assets/60887273/5f8ff455-9ac2-441b-a565-e9898ddc8d65

Usually `^M` indicates, that a carriage return is interpreted as a command. But strangely, that script is already properly formatted to not have `CRLF`. More strangely it only happens on `zsh` and I have never seen such a behavior with any other script I ever used. So the script or something called by it misinterprets the MSYS2 environment somehow.

Either way, this bug is of no real consequence, as I have been compiling my code with emscripten in that environment for years. Just wanted to finally mention it.

**Version of emscripten/emsdk:**
<details><summary>Output of emcc --version</summary>

```console
$ emcc --version
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.38 (9
eff02bc816c50ab0e3b70a3bd5b72a8dc2893a2)
Copyright (C) 2014 the Emscripten authors (see AUTHORS.txt)
This is free and open source software under the MIT license.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR P
URPOSE.
```
</details>

**Comment by sbc100** [comment]
I'm curious if this happens under linux + zsh too?

**Comment by sbc100** [comment]
BTW, presumably you are using the windows version of the SDK binaries?  

This probably relates to the fact that some part of the system is confused about whether you are running on a windows system or a unix system, because in some ways you are running on both, and we don't much testing in these unusual hybrid environments.

Could you also trying running just the `./emsdk construct_env` part which is most of what emsdk_env.sh does?   Does the issue occur when you just run that?

**Comment by sbc100** [comment]
Also, do you know which version of `python` is being used to execute `./emsdk`? 

Can you run `sh -x ./emsdk` to find out?  Is it a windows python.exe or a mingw build of python?

**Comment by FrostKiwi** [comment]
> BTW, presumably you are using the windows version of the SDK binaries?

No. Or at least not intentionally. I followed the official install docs and only did `git clone https://github.com/emscripten-core/emsdk.git` 

> Could you also trying running just the `./emsdk construct_env` part which is most of what emsdk_env.sh does? Does the issue occur when you just run that?

No, the issue does not occur if I run that. It only happens with `source ./emsdk_env.sh`

**Comment by FrostKiwi** [comment]
> Also, do you know which version of `python` is being used to execute `./emsdk`?
> 
> Can you run `sh -x ./emsdk` to find out? Is it a windows python.exe or a mingw build of python?

The MSYS2 one. `Python 3.10.11 (main, Apr  7 2023, 17:44:59)  [GCC 12.2.0 64 bit (AMD64)] on win32` as supplied by the package `mingw-w64-x86_64-python 3.10.11-1` and in path `/mingw64/bin/python`

**Comment by FrostKiwi** [comment]
> Can you run `sh -x ./emsdk` to find out? Is it a windows python.exe or a mingw build of python?

Strangely: `./emsdk activate latest` results in:
```console
$ ./emsdk activate latest
/emsdk activate latest
Resolving SDK alias 'latest' to '3.1.38'
Resolving SDK version '3.1.38' to 'sdk-releases-03ecb526947f6a3702a0d083083799fe410d3893-64bit'
Setting the following tools as active:
   node-15.14.0-64bit
   python-3.9.2-nuget-64bit
   java-8.152-64bit
   releases-03ecb526947f6a3702a0d083083799fe410d3893-64bit
[...]
```
the end of ./emsdk_env.sh results in:
```console
$ source ./emsdk_env.sh
EMSDK_PYTHON = C:/DMGMORI_Development/min/home/artsimow/build/emsdk/python/3.9.2-nuget_64bit/python.exe
JAVA_HOME = C:/DMGMORI_Development/min/home/artsimow/build/emsdk/java/8.152_64bit
(eval):1: command not found: ^M
(eval):1: command not found: ^M
(eval):1: command not found: ^M
(eval):1: command not found: ^M
(eval):1: command not found: ^M
```
but in the end:
```
$ sh -x ./emsdk
+ '[' -z '' ']'
++ dirname ./emsdk
+ PYTHON3=./python/3.9.2-1_64bit/bin/python3
++ dirname ./emsdk
+ PYTHON3_CERT_FILE=./python/3.9.2-1_64bit/lib/python3.9/site-packages/certifi/cacert.pem
+ '[' '!' -f ./python/3.9.2-1_64bit/bin/python3 ']'
++ dirname ./emsdk
+ PYTHON3=./python/3.7.4-2_64bit/bin/python3
++ dirname ./emsdk
+ PYTHON3_CERT_FILE=./python/3.7.4-2_64bit/lib/python3.7/site-packages/certifi/cacert.pem
+ '[' -f ./python/3.7.4-2_64bit/bin/python3 ']'
+ '[' -z '' ']'
++ which python3
+ PYTHON3=/c/DMGMORI_Development/min/mingw64/bin/python3
+ EMSDK_PYTHON=/c/DMGMORI_Development/min/mingw64/bin/python3
+ '[' -z /c/DMGMORI_Development/min/mingw64/bin/python3 ']'
+ exec /c/DMGMORI_Development/min/mingw64/bin/python3 ./emsdk.py
Missing command; Type 'emsdk help' to get a list of commands.
```

**Comment by sbc100** [comment]
> > BTW, presumably you are using the windows version of the SDK binaries?
> 
> No. Or at least not intentionally. I followed the official install docs and the only downloading I did was `git clone https://github.com/emscripten-core/emsdk.git`. So I never chose specific binaries.

But you have no way to run the linux or macOS binaries right?  So you must be using the windows ones I think?  e.g. `emsdk/upstream/bin/clang.exe` exists (after install) and is a windows executable?

**Comment by sbc100** [comment]
Does running emsdk.py with the windows version of python fix the issue?

You can try this in various ways.  One way would be to temporarily remove the mingw64 version of python3.   Another would be to modify the `emsdk` shells script to look for the windows version python instead of the linux on (e.g. `python/3.9.2-nuget_64bit/python.exe` vs `python/3.9.2-1_64bit/bin/python3`.

**Comment by FrostKiwi** [comment]
Huh... no idea why, but on my Windows 11 laptop, the issue is gone, but on my Windows 10 laptop and Windows 11 Workstation the issue remains. All using Windows Terminal 🤔
